assimilation_code/location/threed_cartesian/location_mod.f90

```diff
@@ -272,7 +297,6 @@ function get_dist(loc1, loc2, type1, kind2)
 !> for all combinations of location 1 and location 2 where
```

Comment:
again, a comment for code in the unchanged section.  this isn't your change, but i think the control flow would be clearer if line 285 was an 'endif', not an else.  if no periodic conditions, get distance and return.  period.

then lines 286-288 are no longer inside a conditional section, and the rest of the subroutine deals with the periodic issues.  i'd eventually like to see if we can deal with X, Y, and Z once but for now i believe the code will work with all 8 cases separate and that's fine.

Reply:
I believe we want to keep these lines of code in tact.  That's because the term lines 286 - 288 define the term square_dist, which later referenced in the function when working with the periodic boundary conditions.

Reply:
you're absolutely right those lines have to stay.  it's just now i realize the non-periodic code is about 6 lines and the periodic section is about 100.  i'd like to visually make the two cases more distinct.  i was thinking of something like

```
! the simple non-periodic case
if (.not. any_periodic) then
   if (debug > 0) write(0,*)  'non-periodic distance: '
   get_dist = dist_3d(diff)
   return
endif

! everything below deals with the periodic options
if (debug > 0) write(0,*)  'periodic distance: '
square_dist = dist_3d_sq(diff)
if (debug > 0) write(0,*)  'non-periodic 0 sq_dist, dist: ', square_dist, sqrt(square_dist)

```

Reply:
That is a smart idea, this update looks good to me.  I'll incorporate it within my code.